ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/screen/onboarding/view/OnboardingViewTest.kt

```diff
@@ -239,10 +242,37 @@ class OnboardingViewTest {
         assertEquals(OnboardingStage.Wallet, testSetup.getOnboardingStage())
     }
 
-    private fun newTestSetup(initalStage: OnboardingStage) = TestSetup(composeTestRule, initalStage)
+    @Test
+    @MediumTest
+    fun stage_restoration() {
+        val testSetup = newTestSetup(OnboardingStage.ShieldedByDefault)
+
+        assertEquals(OnboardingStage.ShieldedByDefault, testSetup.getOnboardingStage())
+
+        composeTestRule.onNodeWithText(getStringResource(R.string.onboarding_next)).also {
+            it.performClick()
+        }
+
+        ActivityScenario.launch(MainActivity::class.java).also { scenario ->
```

Comment:
I'm pretty sure this ends up doing something weird.

`ComposeTestRule` (and the compose test framework) bundle a separate ComposeTestActivity (I don't know the exact name but we can find it in AOSP or by looking at the merged manifest) into the test APK.  So all of our tests are run in that special activity, which just sets the view to the Composable we configure.

`ActivityScenario.launch(MainActivity::class.java)` ends up launching a second Activity on top.

The fact that the onboarding stage carries forward to MainActivity after clicking next in ComposeTestRule doesn't immediately make sense to me.

Reply:
Separate from the comment above, I have a concern that this ends up being an integration test that depends on the application having no state on disk (if another test saves a mnemonic, then this test case will fail).

Reply:
Yes, agree, I had also another solution previously, which was probably better. I'll change it.